File: 26. Remove Duplicates from Sorted Array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  if (nums.length < 2) return nums;
  let i = 0;  // Store the last number position
  for (let index = 1; index < nums.length; index++) {
    if (nums[index] != nums[i]) {
      i++;
      nums[i] = nums[index];
    }
  }
  return i + 1;
};

File: 26. Remove Duplicates from Sorted Array/solution.go
func removeDuplicates(nums []int) int {
	if len(nums) <= 1 {
        return len(nums)
    }

    length := 1
    for k, v := range nums {
        if k > 0 && nums[length - 1] != v {
            nums[length] = v
            length++
        }
    }

    return length
}